src/app/modules/time-clock/components/entry-fields/entry-fields.component.spec.ts

```diff
@@ -155,12 +213,37 @@ describe('EntryFieldsComponent', () => {
   it('when a start hour is updated, then dispatch UpdateActiveEntry', () => {
     component.activeEntry = entry ;
     component.setDataToUpdate(entry);
+    const newHour = moment().format('HH:mm:ss');
+    component.entryForm.patchValue({ start_hour : newHour});
     spyOn(store, 'dispatch');
 
     component.onUpdateStartHour();
     expect(store.dispatch).toHaveBeenCalled();
   });
 
+  it('when a start hour is update, then select the last time entry', async(() => {
+    component.activeEntry = entry ;
+    component.setDataToUpdate(entry);
+    const newHour = moment().format('HH:mm:ss');
+
+    component.entryForm.patchValue({ start_hour : newHour});
+    component.onUpdateStartHour();
+
+    expect(component.lastEntry).toBe(state.entries.timeEntriesDataSource.data[1]);
+  }));
+
+  it('when a start hour is updated in other time entry, then dispatch UpdateEntry and UpdateEntryRunning', () => {
```

Comment:
When start_time is updated for a time entry. UpdateEntry and UpdateEntryRuning actions are dispatched.

Reply:
Of course, that phrase is better understood.